js/opening_hours_html_formatter.js

```diff
@@ -0,0 +1,406 @@
+
+/*
+ * Creates a Week object from the given opening ranges.
+ * The opening ranges have to be extracted from an opening_hours.js object.
+ */
+function WeekGenerator() {
+
+    this.getWeek = function(openingRanges) {
+        var openingRangeFormatter = new OpeningRangeFormatter();
+        var week = new Week();
+        if (openingRanges === undefined) {
+            return undefined;
+        }
+        for (var i = 0; i < openingRanges.length; ++i) {
+            var openingRange = openingRanges[i];
+            var day = new Day();
+            var range = new OpeningRange(openingRange, openingRangeFormatter);
+            day.addOpeningRange(range);
+            week.addDay(day);
+        }
+        return week;
+    };
+
+}
+
+if (typeof window.ohhf === "undefined") {
+    window.ohhf = {};
+}
+window.ohhf.WeekGenerator = WeekGenerator;
+
+/*
+ * A week consisting of named days (Monday - Sunday).
+ */
+function Week() {
+
+    this.monday = undefined;
+    this.tuesday = undefined;
+    this.wednesday = undefined;
+    this.thursday = undefined;
+    this.friday = undefined;
+    this.saturday = undefined;
+    this.sunday = undefined;
+
+    this.FIELD_NAMES_ENGLISH = ['sunday', 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday'];
+
+    /*
+     * Adds a Day object.
+     * New and existing days which match by their name are merged.
+     */
+    this.addDay = function(specificDay) {
+        var openingRanges = specificDay.openingRanges;
+        for (var i = 0; i < openingRanges.length; i++) {
+            var openingRange = openingRanges[i];
+            // TODO Get rid of this order dependency.
+            // The array must start with Sunday because Date.getDay() is zero indexed.
+            var dayName = openingRange.getDayName(this.FIELD_NAMES_ENGLISH);
+            if (dayName === "monday") {
+                this.monday = this.getUpdatedDay(this.monday, specificDay);
+            } else if (dayName === "tuesday") {
+                this.tuesday = this.getUpdatedDay(this.tuesday, specificDay);
+            } else if (dayName === "wednesday") {
+                this.wednesday = this.getUpdatedDay(this.wednesday, specificDay);
+            } else if (dayName === "thursday") {
+                this.thursday = this.getUpdatedDay(this.thursday, specificDay);
+            } else if (dayName === "friday") {
+                this.friday = this.getUpdatedDay(this.friday, specificDay);
+            } else if (dayName === "saturday") {
+                this.saturday = this.getUpdatedDay(this.saturday, specificDay);
+            } else if (dayName === "sunday") {
+                this.sunday = this.getUpdatedDay(this.sunday, specificDay);
+            }
+        }
+    };
+
+    /*
+     * Returns an array of days ordered from Monday to Sunday.
+     * A day is only added to the array if it is defined.
+     */
+    this.getDays = function() {
+        var days = [];
+        if (this.monday !== undefined) {
```

Comment:
and this might be replaced with something like
```
this.FIELD_NAMES_ENGLISH.forEach(function(dayname){
  if this.hasOwnProperty(dayname){
    days.push(this[dayname])
  }
})
```
although I am not sure about the scoping of `this` here and have not tested this.